src/apps/js_generic/js_generic.cpp

```diff
@@ -467,8 +467,8 @@ namespace ccfapp
         const std::string path = "/__endpoint__.js";
         JSValue module = JS_Eval(
           ctx,
-          code.c_str(),
-          code.size() + 1,
+          code.data(),
```

Comment:
Does `data()` guarantee that you get a null byte at the end? The size() fix looks fine.

Reply:
Although it looks like quickjs actually doesn't read past the end anyway, probably the docs are slightly incorrect.

Reply:
No, it doesn’t, but if you use c_str/size+1, then the lexer chokes on the null byte, which it doesn’t expect.

This isn’t what the doc suggests, but it’s what the code does.

Reply:
I've reported this to the quickjs-devel list.

Reply:
I've also reverted back to c_str() for consistency with the other calls, even though it doesn't matter here (we pass size()).